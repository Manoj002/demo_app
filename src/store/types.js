export const USER_REGISTERED = 'USER_REGISTERED';
export const SUBMIT_NAME = 'SUBMIT_NAME';
export const NAME_CHANGED = 'NAME_CHANGED';
