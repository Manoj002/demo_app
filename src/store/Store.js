import React, {createContext, useReducer} from 'react';
import {USER_REGISTERED, NAME_CHANGED, SUBMIT_NAME} from './types';

const initialState = {
  isUserRegistered: false,
  userName: undefined,
};

export const StoreContext = createContext(initialState);

export const reducer = (state, action) => {
  switch (action.type) {
    case USER_REGISTERED:
      return {...state, isUserRegistered: action.payload};

    case NAME_CHANGED:
      return {...state, userName: action.payload};

    case SUBMIT_NAME:
      return {...state, userName: action.payload};

    default:
      return state;
  }
};

export const StoreProvider = ({children}) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={[state, dispatch]}>
      {children}
    </StoreContext.Provider>
  );
};
