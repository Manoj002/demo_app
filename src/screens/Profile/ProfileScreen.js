import React, {useContext} from 'react';
import {Text, Button} from '../../commonComponents';
import PropTypes from 'prop-types';
import {StoreContext} from '../../store/Store';
import {
  Container,
  ContentContainer,
  ButtonWrapper,
} from './ProfileScreen.styles';
import Strings from '../../constants/Strings';

const ProfileScreen = ({navigation}) => {
  const [state] = useContext(StoreContext);
  return (
    <Container>
      <ContentContainer>
        <Text
          align="right"
          fontSize={18}
          lineHeight={20}
          color="blue"
          marginRight={20}
          marginBottom={20}>
          {`${state.userName}'s ${Strings.PROFILE}`}
        </Text>
        <Text
          align="center"
          fontSize={18}
          lineHeight={20}
          color="yellow"
          marginBottom={20}>
          {Strings.VARIATIONS_OF_BUTTON}
        </Text>
        <ButtonWrapper>
          <Button buttonAsText>{Strings.PRESS_ME}</Button>
        </ButtonWrapper>
        <ButtonWrapper>
          <Button faded>{Strings.PRESS_ME}</Button>
        </ButtonWrapper>
        <ButtonWrapper>
          <Button colored>{Strings.PRESS_ME}</Button>
        </ButtonWrapper>
        <ButtonWrapper>
          <Button animated>{Strings.SLIDE_ME_TO_CONTINUE}</Button>
        </ButtonWrapper>
        <Text
          align="right"
          marginBottom={20}
          color="blue"
          testID="AccountText"
          onPress={() => navigation.navigate('Account')}>
          {Strings.NAV_ACCOUNT}
        </Text>
        <Text
          align="right"
          color="blue"
          testID="HomeText"
          onPress={() => navigation.navigate('Home')}>
          {Strings.NAV_HOME}
        </Text>
      </ContentContainer>
    </Container>
  );
};

ProfileScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
};

export default ProfileScreen;
