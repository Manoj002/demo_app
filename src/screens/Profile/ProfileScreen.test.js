import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ProfileScreen from './ProfileScreen';
import {StoreProvider, StoreContext} from '../../store/Store';

const navigation = {
  navigate: jest.fn(),
};

const mockValues = {
  isUserRegistered: true,
  userName: 'Manoj',
};

const mockDispatch = jest.fn();

describe('Profile Screen', () => {
  beforeEach(() => jest.clearAllMocks());

  it('renders ProfileScreen correctly', () => {
    const Container = render(
      <StoreProvider>
        <StoreContext.Provider value={[mockValues, mockDispatch]}>
          <ProfileScreen navigation={navigation} />
        </StoreContext.Provider>
      </StoreProvider>,
    );
    expect(Container).toMatchSnapshot();
  });

  it('should navigate to AccountScreen on Text Press', () => {
    const Container = render(
      <StoreProvider>
        <StoreContext.Provider value={[mockValues, mockDispatch]}>
          <ProfileScreen navigation={navigation} />
        </StoreContext.Provider>
      </StoreProvider>,
    );
    const {getByTestId} = Container;
    const AccountText = getByTestId('AccountText');
    fireEvent(AccountText, 'onPress');
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith('Account');
  });

  it('should navigate to HomeScreen on Text Press', () => {
    const Container = render(
      <StoreProvider>
        <StoreContext.Provider value={[mockValues, mockDispatch]}>
          <ProfileScreen navigation={navigation} />
        </StoreContext.Provider>
      </StoreProvider>,
    );
    const {getByTestId} = Container;
    const HomeText = getByTestId('HomeText');
    fireEvent(HomeText, 'onPress');
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith('Home');
  });
});
