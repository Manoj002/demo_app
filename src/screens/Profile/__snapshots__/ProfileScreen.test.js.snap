// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Screen renders ProfileScreen correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#212124",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
          "paddingHorizontal": 12,
          "paddingVertical": 20,
        },
      ]
    }
  >
    <Text
      align="right"
      bold={false}
      color="blue"
      fontSize={18}
      lineHeight={20}
      marginBottom={20}
      marginLeft={0}
      marginRight={20}
      marginTop={0}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontFamily": "Roboto",
            "fontSize": 18,
            "lineHeight": 20,
            "marginBottom": 20,
            "textAlign": "right",
          },
        ]
      }
    >
      Manoj's Profile
    </Text>
    <Text
      align="center"
      bold={false}
      color="yellow"
      fontSize={18}
      lineHeight={20}
      marginBottom={20}
      marginLeft={0}
      marginTop={0}
      style={
        Array [
          Object {
            "color": "#8F9A36",
            "fontFamily": "Roboto",
            "fontSize": 18,
            "lineHeight": 20,
            "marginBottom": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      4 Variations of Button
    </Text>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderRadius": 8,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          align="left"
          bold={false}
          color="blue"
          fontSize={16}
          lineHeight={18}
          marginBottom={0}
          marginLeft={0}
          marginTop={0}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "lineHeight": 18,
                "textAlign": "left",
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#334249",
            "borderRadius": 8,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          align="left"
          bold={false}
          color="blue"
          fontSize={16}
          lineHeight={18}
          marginBottom={0}
          marginLeft={0}
          marginTop={0}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "lineHeight": 18,
                "textAlign": "left",
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#007AFF",
            "borderRadius": 8,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          align="left"
          bold={false}
          color="white"
          fontSize={16}
          lineHeight={18}
          marginBottom={0}
          marginLeft={0}
          marginTop={0}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "lineHeight": 18,
                "textAlign": "left",
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        activeOpacity={0.6}
        marginBottom={0}
        marginTop={0}
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "borderColor": "#343A42",
              "borderRadius": 8,
              "borderWidth": 2,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "backgroundColor": "#007AFF",
              "borderRadius": 8,
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": -2,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
              "width": 50,
            }
          }
        />
        <Text
          align="left"
          bold={false}
          color="blue"
          fontSize={16}
          lineHeight={18}
          marginBottom={0}
          marginLeft={0}
          marginTop={0}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "lineHeight": 18,
                "textAlign": "left",
              },
            ]
          }
        >
          Slide me to continue
        </Text>
      </View>
    </View>
    <Text
      align="right"
      bold={false}
      color="blue"
      fontSize={16}
      lineHeight={18}
      marginBottom={20}
      marginLeft={0}
      marginTop={0}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "lineHeight": 18,
            "marginBottom": 20,
            "textAlign": "right",
          },
        ]
      }
      testID="AccountText"
    >
      Go to Account
    </Text>
    <Text
      align="right"
      bold={false}
      color="blue"
      fontSize={16}
      lineHeight={18}
      marginBottom={0}
      marginLeft={0}
      marginTop={0}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "lineHeight": 18,
            "textAlign": "right",
          },
        ]
      }
      testID="HomeText"
    >
      Go to Home
    </Text>
  </View>
</View>
`;
